frontend/alert/components/AutoComplete.tsx

```diff
@@ -238,12 +159,44 @@ const AutoComplete = ({ defaultValue="", onValueChange, setTimeline, disabled })
     const [backdrop, setBackdrop] = useState("");
 
     const show = active && suggestions.length > 0;
+    const bulkMode = selectedCourses.size > 1;
 
     const setValue = (v) => {
-        onValueChange(v);
         return setInternalValue(v);
     };
 
+    useEffect(() => {
+        // Don't run if first render to prevent defaultValue of value from emptying
+        if (initialRender.current) {
+            initialRender.current = false;
+            return;
+        }
+
+        // update suggestions and input value
+        if (inputRef.current) {
+            if (bulkMode) {
+                // display all sections selected
+                inputRef.current.value = generateCoursesValue();
+            } else if (selectedCourses.size == 1) {
+                // display the one selected section
+                setInternalValue(
+                    selectedCourses.values().next().value.section_id
+                );
+                inputRef.current.value = selectedCourses
+                    .values()
+                    .next().value.section_id;
+            } else if (!inputValDelete.current) {
```

Comment:
I'm probably just misunderstanding something, but why can't we put this logic down in the `onKeyUp` handler of `AutoCompleteInput` [here](https://github.com/pennlabs/penn-courses/pull/344/files#diff-196242b0d979a3169ad3a23cc53cc283728b32d9ce3ec855356930a209d9172cR328) and eliminate the `inputValDelete` ref (which is kind of confusing)?

Reply:
This portion of the code should only be run when the user uncheck the only course they have selected (which causes the value of the input to empty). However, since I am calling `clearSelection` (which modify `selectedCourses`) in `onKeyUp`, this portion of the code will be triggered when it is not suppose to (which causes pressing the backspace to remove the entire value of the input). So I needed to make sure the user is not pressing backspace. Lmk if this make sense lol

Totally agree inputValDelete is confusing, definitely open to any suggestions on how to make this more readable.

Reply:
This makes sense... but maybe passing a `clearInput` function to the `GroupSuggestion` component instead of clearing the input here would allow us to eliminate this confusing code? It makes sense once explained, I'm just worried that someone in the future reading this will have a hard time figuring out what's going on.

Reply:
Oh the thing is I don't want to clear the input when the user is pressing the backspace. I just want to remove the selected course from `selectedCourses`. But doing so triggers the `useEffect` which would clear the input value.

The input value should only clear if the user deselect the only course they have selected (which is why its handled inside the `useEffect`). So I need the `inputValDelete` in the `useEffect` to separate whether it is triggered because the user pressed backspace vs the user deselect the only course they have selected (since both actions modify `selectedCourses` and thus trigger the `useEffect`)

Reply:
But if we passed a `clearInput` function to `GroupSuggestion`, couldn't we handle the case where the user deselects the only course they have selected inside `GroupSuggestion`, and then get rid of these extra cases (for `selectedCourses.size === 0` in the `useEffect`)? That way we wouldn't have to check for `inputValDelete` because these cases wouldn't be handled in the `useEffect`?